packages/mdl-icon-toggle/index.js

```diff
@@ -0,0 +1,148 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import MDLComponent from 'mdl-base';
+import MDLRipple, {MDLRippleFoundation} from 'mdl-ripple';
+
+import MDLIconToggleFoundation from './foundation';
+
+export {MDLIconToggleFoundation};
+
+export default class MDLIconToggle extends MDLComponent {
+  static attachTo(root) {
+    return new MDLIconToggle(root);
+  }
+
+  static buildDom({
+    iconCssClass = 'material-icons',
+    useInnerIconElement = false,
+    toggleOnData = {},
+    toggleOffData = {}
+  } = {}) {
+    const root = document.createElement(useInnerIconElement ? 'span' : 'i');
+    const iconEl = useInnerIconElement ? document.createElement('i') : root;
+    if (iconEl !== root) {
+      iconEl.setAttribute('aria-hidden', 'true');
+      root.appendChild(iconEl);
+    }
+    const {ROOT} = MDLIconToggleFoundation.cssClasses;
+    root.classList.add(ROOT);
+    iconEl.classList.add(iconCssClass);
+    root.setAttribute('role', 'button');
+    root.setAttribute('aria-pressed', 'false');
+    root.tabIndex = 0;
+    root.dataset.toggleOn = JSON.stringify(toggleOnData);
+    root.dataset.toggleOff = JSON.stringify(toggleOffData);
+    if (useInnerIconElement) {
+      root.dataset.iconInnerSelector = `.${iconCssClass}`;
+    }
+
+    return root;
+  }
+
+  constructor() {
+    super(...arguments);
+    this.ripple_ = this.initRipple_();
+  }
+
+  get iconEl_() {
+    const {iconInnerSelector: sel} = this.root_.dataset;
+    return sel ? this.root_.querySelector(sel) : this.root_;
+  }
+
+  initRipple_() {
+    const adapter = Object.assign(MDLRipple.createAdapter(this), {
+      isUnbounded: () => true,
+      isSurfaceActive: () => this.foundation_.isKeyboardActivated(),
+      computeBoundingRect: () => {
+        const dim = 48;
+        const {left, top} = this.root_.getBoundingClientRect();
+        return {
+          left,
+          top,
+          width: dim,
+          height: dim,
+          right: left + dim,
+          bottom: left + dim
+        };
+      }
+    });
+    const foundation = new MDLRippleFoundation(adapter);
+    return new MDLRipple(this.root_, foundation);
+  }
+
+  destroy() {
+    this.ripple_.destroy();
+    super.destroy();
+  }
+
+  getDefaultFoundation() {
+    return new MDLIconToggleFoundation({
+      addClass: className => this.iconEl_.classList.add(className),
+      removeClass: className => this.iconEl_.classList.remove(className),
+      registerInteractionHandler: (type, handler) => this.root_.addEventListener(type, handler),
+      deregisterInteractionHandler: (type, handler) => this.root_.removeEventListener(type, handler),
+      setText: text => {
+        this.iconEl_.textContent = text;
+      },
+      getTabIndex: () => /* number */ this.root_.tabIndex,
+      setTabIndex: tabIndex => {
```

Comment:
eslint complains about this, and rightfully so:

<img width="794" alt="screen shot 2016-09-21 at 14 59 36" src="https://cloud.githubusercontent.com/assets/1185269/18724893/12d86d42-800c-11e6-8892-a6b7bd3cd489.png">

This will actually return a value as the result of an expression. Although it doesn't make a difference, it seems more straightforward to explicitly wrap this statement in a BlockExpression. If you disagree, we can disable the rule?